File: 1646. Kth Missing Positive Number/Solution.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        
        vector<int> B;
        for(int i=0;i<arr.size();i++){
            B.push_back(arr[i]-i-1);
            cout<<B[i]<<endl;
        }
        int low =0,high=arr.size();
        while(low<high){
            int mid = low + (high-low)/2;
            if(B[mid]>=k){
                high = mid;
            } else {
                low = mid+1;
            }
        }
        //cout<<low<<"-"<<high<<endl;
        return high+k;
    }
};

File: 1646. Kth Missing Positive Number/Solution.py
class Solution(object):
    def findKthPositive(self, arr, k):
        """
        :type arr: List[int]
        :type k: int
        :rtype: int
        """
        i, cnt = 1, 0
        while cnt<k:
            if i not in arr:
                cnt+=1
            if cnt==k:
                return i
            i+=1
        return i
           
        